### ISSUE TITLE
Rule Documentation UX improvement suggestions: navigation + link to source

### ISSUE THREAD

**Issue by sbrugman** [astral-sh/ruff#13367]
Hi there,

As user I frequently wish the docs would have these functionalities:

## Navigation between rule pages

Navigate to another rule of the same prefix, e.g. on any rule page https://docs.astral.sh/ruff/rules/suppressible-exception/ it would be great to be able to navigate to the previous/next rule with the same prefix. This would be a nicer experience to explore  a set of rules then going back and opening each rule in a new tab from the (huge) all rules page.

One option would be to enable footer navigation: https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/#navigation

Alternatively, I could imagine:

- being able to navigate back from the rule page, e.g. making "Derived from the XXX linter." clickable link.
- a navbox with the rules on the bottom of each rule page (like wikipedia has, personal favourite option):

<img width="800" src="https://github.com/user-attachments/assets/6ccc2acd-a69f-4679-96a3-3283fd4f160a">

## Link to source/fixtures

Rule documentation as is does not answer questions such as:
- Rule X violation detected, but I suspect it's a false positive, what exact cases does this rule match?
- I would expect Rule Y to trigger, but it did not, is that on purpose?

To allow the user to more easily answer these questions before opening an issue, it would be helpful to link the source code and/or the Python fixtures for a rule on that page. Going from the root directory on GitHub, this can save around 7 clicks, e.g. https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/resources/test/fixtures/flake8_tidy_imports/TID251.py.

**Comment by sbrugman** [comment]
Example of what the navbox approach would look like:

<img src="https://github.com/user-attachments/assets/806d5490-b8a5-4a10-acde-5e97fa5cf6e2">

**Comment by MichaReiser** [comment]
I do find the navbox very verbose and hard to parse. It also lacks any relevant description or additional information where it is unclear to me what the advantage is over just having the information in the rules table.

**Comment by sbrugman** [comment]
I do agree. What about the alternative of enabling "next" and "previous" navigation for rules?

<img src="https://github.com/user-attachments/assets/c3e7909c-b7f6-4cc5-a53d-47949deafd3a">

**Comment by MichaReiser** [comment]
> I do agree. What about the alternative of enabling "next" and "previous" navigation for rules? (like the uv docs has. [pylint](https://pylint.readthedocs.io/en/stable/user_guide/messages/error/assigning-non-slot.html) does it too)

I'm open to that, although it's unclear to me if there's a natural order for reading the rules (from start to end). How would next and previous work? Would it only navigate between rules of the same category, or is it cross-categories (the next button of a category last rule goes to the next category's first rule)?
